### ISSUE TITLE
Flask Benchmark

### ISSUE THREAD

**Issue by neurosnap** [Eyepea/API-Hour#2]
https://github.com/Eyepea/API-Hour/blob/master/benchmarks/flask/application.py

Is there any reason specifically why we are using psycopg directly instead of using something like Flask-SQLAlchemy (http://pythonhosted.org/Flask-SQLAlchemy/)?

It's pretty much the standard for interacting with PostgreSQL and Flask-SQLAlchemy is almost always the preferred method when in the wild.  I can whip one up really quickly if we want to add another benchmark into the mix.

**Comment by GMLudo** [comment]
Hi,

If you want to add an URL in Flask application to use sqlalchemy, why not, I can benchmark that.

We don't use ORM because:
1. With `psycopg2.extras.RealDictCursor` we retrieve a dict for each record, it's enough to access easily to columns
2. We improve performances to avoid a layer
3. We do very complex queries (until 100 lines of SQL) to generate quickly statistics or for BI. We need more time to write that with ORM instead of SQL, and in some cases, it's impossible with ORM.
4. We use advanced features of PostgreSQL like `jsonb` or `WITH` statement. Again, we save our time to write SQL directly.
5. When we have an issue, it's only a SQL issue, not also a ORM issue.

In short, for economical reasons, we don't use ORM.

Regards.

**Comment by neurosnap** [comment]
Thanks for the quick response, it makes a lot of sense.

I was thinking more about the benchmarking of Flask, adding an ORM certainly comes with overhead, but it's how most people interact with PostgreSQL, which is what I'm mostly interested in.

Anyway, let me know if you need anymore help, happy to lend a hand.

**Comment by GMLudo** [comment]
If you want to propose a pull request with Flask+sqlalchemy, it's ok for me, even if we should attract some haters from sqlachemy community with the next benchmark, after flask and django communities ;-)
In the same time, it should be interesting to show the "cost" of an ORM.
But, please, your sqlalchemy should generate the same SQL query I use in my benchmark to compare apples with apples.
I've tried with Django-ORM, but finally, I didn't success.
